crates/yas_core/src/libraries/tick_bitmap.cairo

```diff
@@ -146,14 +146,18 @@ mod TickBitmap {
         let divisor = IntegerTrait::<i32>::new(256, false);
         let negative_lower_bound = IntegerTrait::<i32>::new(256, true);
 
-        let result = if tick >= zero && tick <= upper_bound {
+        let result = if tick >= zero && tick <= upper_bound { //tick: [0, 255]
             zero
-        } else if tick > upper_bound {
+        } else if tick > upper_bound { //tick: [256, +inf)
             tick / divisor
-        } else if tick >= negative_lower_bound {
+        } else if tick >= negative_lower_bound { //tick: [-256, -1]
             one_negative
-        } else {
-            tick / divisor + one_negative
+        } else { //(-inf, -257]
+            if (mod_i32(tick, divisor) != zero) {
+                IntegerTrait::<i32>::new((tick.mag / divisor.mag) + 1, true)
+            } else {
+                IntegerTrait::<i32>::new((tick.mag / divisor.mag), true)
```

Comment:
Why this behavior should be here and not at the library level?

Reply:
in Uni, this calculate_word(tick) function is actually tick >> 8, effectively acting as a division "rounded towards 0".
In our i32 library (which has the same behaviour as the latest Orion lib), when dividing it rounds towards the nearest int ( not towards -Inf as i thought before! ).
So, this code is to emulate the "tick >> 8" (or, division and rounding towards 0) behaviour.
Alternately, I imagine, we could implement a bit-operations library, to perform bit shifting operations with it directly.

Reply:
I like the idea of including this operation directly in the library for the future. Could you add a ToDo?